crates/control/src/calibration_controller.rs

```diff
@@ -0,0 +1,299 @@
+                camera,
+                target,
+                dispatch_time,
+            },
+            None => {
+                info!("\tNothing else to LOOKAT, goto PROCESS");
+                CalibrationCommand::PROCESS
+            }
+        }
+    }
+
+    fn get_next_look_at(&mut self) -> Option<(Point2<Ground>, Option<CameraPosition>)> {
+        let current_index = self.current_look_at_index;
+        self.current_look_at_index += 1;
+        if current_index < self.look_at_list.len() {
+            Some(self.look_at_list[current_index])
+        } else {
+            None
+        }
+    }
+}
+
+fn collect_filtered_values(
+    measurement_perception_input: &PerceptionInput<Vec<&CalibrationCaptureResponse<Measurement>>>,
+    original_dispatch_time: &CycleTime,
+) -> (Option<Measurement>, bool) {
+    let result = measurement_perception_input
+        .persistent
+        .iter()
+        .flat_map(|(_cycle_timestamp, measurements)| measurements.iter())
+        .find(|&measurement| match measurement {
+            CalibrationCaptureResponse::CommandRecieved {
+                dispatch_time,
+                output: _,
+            } => original_dispatch_time.start_time == dispatch_time.start_time,
+            CalibrationCaptureResponse::Idling => false,
+            CalibrationCaptureResponse::RetriesExceeded { dispatch_time } => {
+                original_dispatch_time.start_time == dispatch_time.start_time
+            }
+        });
+
+    let mut retries_exceeded = false;
+    let mut value_found = false;
+    let measurement: Option<Measurement> = result.and_then(|value| match value {
+        CalibrationCaptureResponse::Idling => None,
+        CalibrationCaptureResponse::CommandRecieved {
+            dispatch_time: _,
+            output,
+        } => {
+            if output.is_some() {
+                value_found = true;
+                output.clone()
+            } else {
+                None
+            }
+        }
+        CalibrationCaptureResponse::RetriesExceeded { dispatch_time: _ } => {
+            retries_exceeded = true;
+            None
+        }
+    });
+
+    (measurement, retries_exceeded || value_found)
+}
+
+// TODO Add fancier logic to either set this via parameters OR detect the location, walk, etc
+fn generate_look_at_list() -> Result<Vec<(Point2<Ground>, Option<CameraPosition>)>> {
+    let look_at_points: Vec<Point2<Ground>> = vec![
+        point!(1.0, 0.0),
+        point!(1.0, -0.5),
+        point!(3.0, -0.5),
+        point!(3.0, 0.0),
+        point!(3.0, 0.5),
+        point!(1.0, -0.5),
+    ];
+
+    let attach_camera_to_lookat =
+        |point: &Point2<Ground>,
+         camera_position: &CameraPosition|
+         -> (Point2<Ground>, Option<CameraPosition>) { (*point, Some(*camera_position)) };
+
+    Ok(look_at_points
+        .iter()
+        .map(|&point| attach_camera_to_lookat(&point, &CameraPosition::Top))
+        .collect_vec())
```

Comment:
You don't need the specialization here.
```suggestion
        .collect())
```

Reply:
Ah yes,